src/actions/expressions.js

```diff
@@ -37,26 +38,30 @@ export function addExpression(input: string) {
       return dispatch(evaluateExpression(expression));
     }
 
-    dispatch({
-      type: "ADD_EXPRESSION",
-      input
-    });
+    dispatch({ type: "ADD_EXPRESSION", input, expressionError });
 
     const newExpression = getExpression(getState(), input);
-    dispatch(evaluateExpression(newExpression));
+    if (newExpression) {
```

Comment:
is it possible to add expression and not have one?

do we need the IF?

Reply:
The line that was guarded was producing a type error when the guard wasn't present.